### ISSUE TITLE
Automate the bill of materials

### ISSUE THREAD

**Issue by lamara11** [amosproj/amos2024ss01-xcelerator-demo-app#52]
## User story

1. As a developer
2. I want to automate the bill of materials (BOM) generation process 
3. So that I can efficiently track and manage the components and dependencies used in our projects.

## Acceptance criteria

- [x] The automated bill of materials accurately captures all project dependencies, including component names and versions.
- [x] Capturing only first level dependencies is sufficient. 

## Definition of done (DoD)

- [x] Added only after week 5
- [x] The same for all features
- [x] Here goes the project specific part

## DoD general criteria

- [x] Feature has been fully implemented
- [x] Feature has been merged into the mainline
- [x] All acceptance criteria were met
- [x] Product owner approved features
- [x] All tests are passing
- [x] Developers agreed to release
